### ISSUE TITLE
Prevent "flashing" of screen on lock/unlock

### ISSUE THREAD

**Issue by draoncc** [google/xsecurelock#94]
Would be cool if there was a way to prevent the "flashing" of a blank window on locking while the saver isn't done drawing on it yet.

Perhaps on locking let the saver determine when the locking process continues, so that it may finish possible drawing processes on $XSCREENSAVER_WINDOW?

**Comment by divVerent** [comment]
You mean black flashing? Hm... would be cool to fix, but we currently permit any xscreensaver savers and there is no real way to detect when they are done. Also, before the window is visible, applications cannnot draw on it.

**Comment by divVerent** [comment]
Closing for now; patch welcome though.

**Comment by vincentbernat** [comment]
I have looked a bit at this. The problem is quite visible because of the 1-second delay before starting children. Just moving the delay at the end of the loop helps. I have also tried to move window mapping just before the loop and sleeping after running children before the loop and it almost solves the issue (when not using the multiplexer). This is not perfect yet and I don't know who is responsible for the last shade of black (maybe that's my own app as it has to draw the pixmap once it is exposed).

I have also tried to use a 32-bit visual and make the background transparent, but I did not succeed. Not sure if I am missing a window or if I don't do the right thing. I'll investigate a bit more.
